app/controllers/api/v2/compliance/policies_controller.rb

```diff
@@ -69,7 +69,13 @@ def create
       param_group :policy
 
       def update
-        process_response @policy.update_attributes(policy_params)
+        ::ForemanOpenscap::Policy.transaction do
```

Comment:
could you pleace extract this logic to model's method, somehing like

```ruby
def change_policy_deploy_type(params)
  self.class.transaction do
    ...
  end
end
```

and then reuse it in both controllers? it needs to return true/false and base on that, we'd process_response or process_success/process_error